src/containers.jl

```diff
@@ -165,31 +178,33 @@ function update_ref!(
 )
     if !isnothing(ref_state)
         filtered.particles[1] = ref_state[step]
+        # TODO: is this correct?
+        if step > 0
+            filtered.log_weights[1] = 0
+        end
```

Comment:
In the Chopin code you shared, this step isn't there so probably not. 

The tests still pass without it so I'll assume it's not needed until I think otherwise.